src/mscorlib/corefx/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -179,8 +195,8 @@ private static bool StringEqualsAscii(string s1, string s2, bool ignoreCase = tr
 
             for (int i = 0; i < s1.Length; i++)
             {
-                char c1 = ignoreCase ? TextInfo.ChangeCaseAscii(s1[i]) : s1[i];
-                char c2 = ignoreCase ? TextInfo.ChangeCaseAscii(s2[i]) : s2[i];
+                char c1 = ignoreCase ? ChangeCaseAscii(s1[i]) : s1[i];
+                char c2 = ignoreCase ? ChangeCaseAscii(s2[i]) : s2[i];
```

Comment:
I wonder whether this loop should be:
```C#
if (ignoreCase)
{
    for (int i = 0; i < s1.Length; i++)
        if (ChangeCaseAscii(s1[i]) != ChangeCaseAscii(s2[i]))
            return false;
}
else
{
    for (int i = 0; i < s1.Length; i++)
        if (s1[i] != s2[i])
            return false;
}
```
but maybe the compiler(s) do the appropriate optimizations without that?